lib/Sema/ConstraintSystem.h

```diff
@@ -3371,32 +3384,44 @@ Expr *simplifyLocatorToAnchor(ConstraintSystem &cs, ConstraintLocator *locator);
 /// Common interface to encapsulate attempting choices of
 /// different entities, such as type variables (types)
 /// or disjunctions (their choices).
-struct TypeBinding {
+class TypeBinding {
+  unsigned Index;
+
+public:
+  TypeBinding(unsigned index) : Index(index) {}
+
   virtual ~TypeBinding() {}
   virtual void attempt(ConstraintSystem &cs) const = 0;
+
+  /// Position of this binding in the chain.
+  unsigned getIndex() const { return Index; }
+
+  virtual bool isDisabled() const { return false; }
```

Comment:
Actually let me try to convert this into flags of the binding, so I don't have to create so many virtual methods...

Reply:
Okay, all done!